### PR TITLE
native fixes

### PR DESCRIPTION
**Pull request by rurirei**
1. manifest: extractNativeLibs = true
2. native: call SupportSet.Shutdown() when stopService
3. native: call runTun2socks() after vpn.Setup() ~~https://github.com/2dust/v2rayNG/pull/635#pullrequestreview-501540965~~ https://github.com/2dust/v2rayNG/issues/636
4. vpnservice: call sendFd() before tun2socks start
5. native: remove senfFd in Escorting

### CODE REVIEW

File: V2rayNG/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -27,6 +27,7 @@
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:supportsRtl="true"
+        android:extractNativeLibs="true"
```
Comment (yuhan6665) [inline V2rayNG/app/src/main/AndroidManifest.xml:30]: Are you sure this is needed?

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt [hunk 2]
```diff
@@ -189,7 +191,7 @@ class V2RayVpnService : VpnService(), ServiceControl {
                 break
             } catch (e: Exception) {
                 Log.d(packageName, e.toString())
-                if (tries > 5) break
+                if (tries > 20) break  // below 10 as my experience
```
Comment (yuhan6665) [inline V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt]: this comment is confusing

File: AndroidLibV2rayLite/interact.go [hunk 3]
```diff
@@ -171,6 +160,17 @@ func (v *V2RayPoint) pointloop() error {
 	v.SupportSet.Prepare()
 	v.SupportSet.Setup(v.status.GetVPNSetupArg(v.EnableLocalDNS, v.ForwardIpv6))
 	v.SupportSet.OnEmitStatus(0, "Running")
+
+	if err := v.runTun2socks(); err != nil {
+		log.Println(err)
+		return err
+	}
+
+	log.Printf("EnableLocalDNS: %v\nForwardIpv6: %v\nDomainName: %s",
```
Comment (yuhan6665) [inline AndroidLibV2rayLite/interact.go:168]: Usually the log statement is at the beginning

**Review by yuhan6665 — COMMENTED** [review]
Generally looks ok

File: V2rayNG/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (rurirei) [reply V2rayNG/app/src/main/AndroidManifest.xml:30]: When building the release version of your app, package uncompressed .so files in the APK by setting android:extractNativeLibs="false" in the <application> element of your app's manifest. Disabling this flag prevents PackageManager from copying .so files from the APK to the filesystem during installation and has the added benefit of making updates of your app smaller. When building your app using Android Gradle plugin 3.6.0 or higher, the plugin sets this property to "false" by default. https://developer.android.com/topic/performance/reduce-apk-size#extract-false

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt (hunk 2 above)
Comment (rurirei) [reply V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt]: sendFd() should only called before tun2socks start, and tries times is below 10 as my experience. call sendFd() after tun2socks started ( inside EscortRun()) causes IOException https://github.com/2dust/v2rayNG/issues/636#issuecomment-703487440 , so 6th commit remove this call.

File: V2rayNG/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (rurirei) [reply V2rayNG/app/src/main/AndroidManifest.xml:30]: sure. we needs to run binarys inside nativeDir. 
>> When building the release version of your app, package uncompressed .so files in the APK by setting android:extractNativeLibs="false" https://developer.android.com/topic/performance/reduce-apk-size#extract-false

File: AndroidLibV2rayLite/interact.go (hunk 3 above)
Comment (rurirei) [reply AndroidLibV2rayLite/interact.go:168]: run tun2socks at the last.

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt [hunk 4]
```diff
@@ -193,7 +194,7 @@ class V2RayVpnService : VpnService(), ServiceControl {
                 break
             } catch (e: Exception) {
                 Log.d(packageName, e.toString())
-                if (tries > 20) break  // below 10 as my experience
+                if (tries > 5) break  // waits below 2 seconds as my experience
```
Comment (yuhan6665) [inline V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt:197]: Looks better now

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt [hunk 5]
```diff
@@ -161,11 +161,8 @@ object V2rayConfigUtil {
 //                v2rayConfig.inbounds.add(httpCopy)
 //            }
 
-            if (!app.defaultDPreference.getPrefBoolean(SettingsActivity.PREF_LOCAL_DNS_ENABLED, false)) {
-                v2rayConfig.inbounds[0].sniffing?.enabled = true
-            } else {
-                v2rayConfig.inbounds[0].sniffing?.enabled = app.defaultDPreference.getPrefBoolean(SettingsActivity.PREF_SNIFFING_ENABLED, true)
-            }
+            // enforce trafficSniffing to prevent from dns-poisoning
+            v2rayConfig.inbounds[0].sniffing?.enabled = true
```
Comment (yuhan6665) [inline V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt:165]: this doesn't make sense. Sending dns traffic to server also mitigate dns-poisoning. I think someone may not want sniffing function.

**Review by yuhan6665 — COMMENTED** [review]
Please keep the pr commits clean. Use rebase instead of merge. If a previous commit need amend, use force-push to change it. (Instead of adding another revert commit)

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt (hunk 5 above)
Comment (rurirei) [reply V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt:165]: sniffing had to enabled (for VPN mode), or you will met dns-poisoning, you can try it.

**Comment by rurirei** [comment]
will clean up later.

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt (hunk 5 above)
Comment (rurirei) [reply V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt:165]: https://github.com/2dust/v2rayNG/issues/653

**Pull request closed without merging**